responses/16-dir-targets.md

```diff
@@ -5,92 +5,62 @@ You might have a project where there is a directory :file_folder: with a collect
+
+list(
+  tar_target(
+    plot_data, 
+    combine_into_df(c('1_fetch/in/file1.csv',
+                      '1_fetch/in/file2.csv',
+                      '1_fetch/in/file3.csv'))
+  ),
+  tar_target(figure_1_png, my_plot(plot_data))
+)
 ```
 
-:point_up: This coding would work, as it tells the dependency manager which files to track for changes, and where the files in the directory are. But this solution is less than ideal, both because it doesn't scale well to **many** files, and because it doesn't adapt to new files coming into the `1_fetch/in` directory :file_folder: (the pipeline coder needs to manually add file names)
-
----
-
-Alternatively, what about adding the directory as an input to the recipe, like this (you'd also need to modify your `combine_into_df` function to use `dir(work_dir)` to generate the file names)?
-```yaml
-sources:
-  - combine_files.R
-
-targets:
-  all:
-    depends: figure_1.png
-
-  plot_data:
-    command: combine_into_df(work_dir = '1_fetch/in')
-    
-  figure_1.png:
-    command: my_plot(plot_data)
-```
-
-After running `scmake()`, :point_up: this looks like it works, but results in a rather cryptic warning message might end up being ignored:
-```
-Warning messages:
-1: In structure(.Call(C_Rmd5, files), names = files) :
-  md5 failed on file '1_fetch/in'
+:point_up: But this coding wouldn't work, as it only tells the dependency manager to track the filepaths as a character vector and doesn't tell it to track changes in the files themselves. So, the only way it would rebuild is if the filepaths themselves changed. So, we need to recode so that `targets` tracked those as files by creating a target whose command returned a vector of filepaths and included the argument `format = "file"` (recall that we learned a bit about input files as targets in [the previous section]({{ store.undoc_file_url }})). 
+
+```r
+library(targets)
+source("combine_files.R")
+
+list(
+  tar_target(in_files, 
+             c('1_fetch/in/file1.csv',
+               '1_fetch/in/file2.csv',
+               '1_fetch/in/file3.csv'), 
+             format = "file"),
+  tar_target(
+    plot_data, 
+    combine_into_df(in_files)
+  ),
+  tar_target(figure_1_png, my_plot(plot_data))
+)
 ```
 
-An explaination of that warning message - In order to determine if file contents have changed, remake uses `md5sum()` from the `tools` package to create a unique hash for each target. You can take a look at what this does:
-```
-tools::md5sum('3_visualize/out/figure_1.png')
-      3_visualize/out/figure_1.png 
-"500a71be79d1e45457cdb2c31a03be46" 
-```
-`md5sum()` fails when you point it toward a directory, since it doesn't know what to do with that kind of input. 
+While this solution would work, it is less than ideal because it doesn't scale well to **many** files, nor would it adapt to new files coming into the `1_fetch/in` directory :file_folder: (the pipeline coder would need to manually add file names to the `in_files` target).
```

Comment:
Excellent point. I think I like pointing out the pattern of listing individual filepaths as inputs and then pointing out that this approach won't scale well. I think it is a nice transition between the last section and this one. The example that is written incorrectly can be deleted because there isn't much use in showing it.